python/nemo_relay/integrations/langchain/callbacks.py

```
@@ -218,6 +218,26 @@ def _close_completed_scopes_locked(self) -> None:
                     metadata=completed.metadata,
                     timestamp=completed.ended_at,
                 )
+            except ValueError:
+                # ``scope.pop`` converts output before mutating the stack, so a
+                # rejected payload can be safely discarded and the close retried.
+                if completed.output is None:
+                    _logger.error("NeMo Relay: scope.pop failed", exc_info=True)
```

Ruff (0.16.1)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)